server/server_test.go

```diff
@@ -1894,25 +1936,32 @@ func getStmtCnt(content string) (stmtCnt map[string]int) {
 
 const retryTime = 100
 
-func (cli *testServerClient) waitUntilServerCanConnect() {
+func (cli *testServerClient) waitUntilCustomizationServerCanConnect(dsn string) {
 	// connect server
 	retry := 0
 	for ; retry < retryTime; retry++ {
 		time.Sleep(time.Millisecond * 10)
-		db, err := sql.Open("mysql", cli.getDSN())
+		db, err := sql.Open("mysql", dsn)
 		if err == nil {
-			err = db.Close()
-			if err != nil {
-				panic(err)
+			succeed := db.Ping() == nil
+			if err = db.Close(); err != nil {
+				log.Error("fail to connect db", zap.String("err", err.Error()))
```

Comment:
IMHO it's better to put dsn in the log, it would be helpful if we meet `fail to connect db` in the future.

Reply:
Have added.